packages/snap/src/lib/keyDeriver.ts

```diff
@@ -22,36 +22,82 @@ async function getKeyDeriver(chain: ChainDefinition) {
+ * Derive an Antelope private key from the key tree at the given address index.
  *
- * @param addressIndex - The index of the address to derive.
- * @returns The public key.
+ * @param keyIndex - The index of the key to derive.
+ * @returns The private key.
  * @throws If the key tree is not initialized.
  */
-export async function derivePublicKey(
+async function derivePrivateKey(
+  chain: ChainDefinition,
+  keyIndex = 0,
+): Promise<PrivateKey> {
+  const keyDeriver = await getKeyDeriver(chain);
+  const derived = await keyDeriver(keyIndex);
+
+  if (!derived.privateKeyBytes) {
+    throw new Error('Private key not found');
+  }
+
+  return new PrivateKey(KeyType.K1, Bytes.from(derived.privateKeyBytes));
+}
+
+/**
+ * Derive an Antelope owner public key from the key tree.
+ * Always uses index 0 which is reserved for owner permissions.
+ *
+ * @returns The owner public key.
+ * @throws If the key tree is not initialized.
+ */
+export async function deriveOwnerPublicKey(
   chain: ChainDefinition,
-  addressIndex = 0,
 ): Promise<PublicKey> {
-  return (await derivePrivateKey(chain, addressIndex)).toPublic();
+  return (await deriveOwnerPrivateKey(chain)).toPublic();
 }
 
 /**
- * Derive an Antelope private key from the key tree at the given address index.
+ * Derive an Antelope active public key from the key tree.
+ * Cannot use index 0 which is reserved for owner permissions.
  *
- * @param addressIndex - The index of the address to derive.
- * @returns The private key.
+ * @param keyIndex - The index of the address to derive (must not be 0).
+ * @returns The active public key.
+ * @throws If the key tree is not initialized or if index 0 is used.
+ */
+export async function deriveActivePublicKey(
+  chain: ChainDefinition,
+  keyIndex = 1,
+): Promise<PublicKey> {
+  return (await deriveActivePrivateKey(chain, keyIndex)).toPublic();
+}
+
+/**
+ * Derive an Antelope owner private key from the key tree.
+ * Always uses index 0 which is reserved for owner permissions.
+ *
+ * @returns The owner private key.
  * @throws If the key tree is not initialized.
  */
-export async function derivePrivateKey(
+export async function deriveOwnerPrivateKey(
   chain: ChainDefinition,
-  addressIndex = 0,
 ): Promise<PrivateKey> {
-  const keyDeriver = await getKeyDeriver(chain);
-  const derived = await keyDeriver(addressIndex);
+  return derivePrivateKey(chain, 0);
+}
 
-  if (!derived.privateKeyBytes) {
-    throw new Error('Private key not found');
+/**
+ * Derive an Antelope active private key from the key tree.
+ * Cannot use index 0 which is reserved for owner permissions.
+ *
+ * @param keyIndex - The index of the address to derive (must not be 0).
+ * @returns The active private key.
+ * @throws If the key tree is not initialized or if index 0 is used.
+ */
+export async function deriveActivePrivateKey(
+  chain: ChainDefinition,
+  keyIndex = 1,
+): Promise<PrivateKey> {
+  if (keyIndex === 0) {
```

Comment:
I don't think this is a strong enough check.

If I pass in `keyIndex = undefined` to this function, the value won't be equal to 0 - so it'll bypass this and then pass `undefined` to `derivePrivateKey`, which will default to 0 and return the owner key.

Maybe we need to write it like this:

```ts
if (Number(keyIndex) > 0) {
  return derivePrivateKey(chain, keyIndex);
}
throw new Error('Invalid key index');
```

We just need to make sure there's no way to access index 0 from this call.

Reply:
if you pass `undefined` then the default `1` will be used 😄

Reply:
if you want, I'll change the default of `derivePrivateKey` to 1 so the owner key isn't returned by default

Reply:
Done 👍